### PR TITLE
created substructure_search function

### PR DESCRIPTION
**Pull request by maneav78**
Homework 1(4)

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,2 +1,29 @@
...
+    if substructure is None:
+        raise ValueError(f'Invalid substructure SMILES: {mol}')
+    
+    substructure_num_atoms = substructure.GetNumAtoms()
+
+    matches = []
+    for molecule in mols:
+        object_mol = Chem.MolFromSmiles(molecule)
+        if object_mol is None:
+            print(f'Skipping invalid molecule SMILES: {molecule}')
```
Comment (JDima) [inline src/main.py]: It is better to have the same approach to processing incorrect molecules, as done above

File: src/main.py [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+            continue 
+        
+        if object_mol.HasSubstructMatch(substructure):
+           matches.append(molecule)
+
+    return matches
+
+# testing
+if name == "main":
+    print(substructure_search(["CCO", "c1ccccc1", "CC(=O)O", "CC(=O)Oc1ccccc1C(=O)O"], "c1ccccc1"))
```
Comment (JDima) [inline src/main.py]: It will be better to use `assert` function to check than to check your screen.

**Comment by JDima** [comment]
Everything is done right. You can merge to main.

File: src/main.py (hunk 1 above)
Comment (maneav78) [reply src/main.py]: Done!

File: src/main.py (hunk 2 above)
Comment (maneav78) [reply src/main.py]: Also done!

**Pull request merged**